platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java

```diff
@@ -67,10 +68,12 @@ protected SocketConnection(
             final boolean outbound,
             final Socket socket,
             final SyncInputStream dis,
-            final SyncOutputStream dos) {
+            final SyncOutputStream dos,
+            final Configuration configuration) {
```

Comment:
Similar to the reconnect classes from earlier in the PR, I suggest passing in a context (as the first argument) and then unpacking the configuration.